common/network-common/src/main/java/org/apache/spark/network/util/NettyUtils.java

```diff
@@ -36,6 +36,22 @@
  * Utilities for creating various Netty constructs based on whether we're using EPOLL or NIO.
  */
 public class NettyUtils {
+
+  /**
+   * Specifies an upper bound on the number of Netty threads that Spark requires by default.
+   * In practice, only 2-4 cores should be required to transfer roughly 10 Gb/s, and each core
+   * that we use will have an initial overhead of roughly 32 MB of off-heap memory, which comes
+   * at a premium.
+   *
+   * Thus, this value should still retain maximum throughput and reduce wasted off-heap memory
+   * allocation. It can be overridden by setting the number of serverThreads and clientThreads
+   * manually in Spark's configuration.
+   */
+  private static int MAX_DEFAULT_NETTY_THREADS = 8;
+
+  private static PooledByteBufAllocator[] _sharedPooledByteBufAllocator =
```

Comment:
Why not just two fields? their names could make their use clearer. Or else if there's a need for an array it can be final.

Reply:
As only one method reads/writes this `_sharedPooledByteBufAllocator` I keep them together in this array, so I add the `final`.

Reply:
Why an array though? vs just two simple fields?

Reply:
It is a bit simple to use and array as based on the index I can set the right element. 
Does not matter which one (0 or 1) as long as it is consistent (and that one line of code calculating the index is consistent).

Otherwise an extra if should be introduced to set the right field. Whit the checking of is it initialised it would be a scoped if. And I just prefer to avoid scoped if-s, like:

```scala
  public static synchronized PooledByteBufAllocator getSharedPooledByteBufAllocator(
      boolean allowDirectBufs,
      boolean allowCache) {
    if (allowCache) {
      if (_sharedPooledByteBufAllocatorWithAllowCache == null) {
        _sharedPooledByteBufAllocatorWithAllowCache =
            createPooledByteBufAllocator(
                allowDirectBufs,
                allowCache,
                defaultNumThreads(0));
      }
      return _sharedPooledByteBufAllocatorWithAllowCache;
    } else {
      if (_sharedPooledByteBufAllocatorWithoutAllowCache == null) {
        _sharedPooledByteBufAllocatorWithoutAllowCache =
            createPooledByteBufAllocator(
                allowDirectBufs,
                allowCache,
                defaultNumThreads(0));
      }
      return _sharedPooledByteBufAllocatorWithoutAllowCache;
    }
  }
```
Or am I miss something?

Reply:
I guess you could check `allowCache` twice to figure out what reference to check for null, and which to update. OK I don't feel strongly about it.